tracing-subscriber/src/registry/mod.rs

```diff
@@ -294,15 +367,69 @@ where
 
     /// Returns the ID of this span's parent, or `None` if this span is the root
     /// of its trace tree.
+    #[deprecated(
+        note = "this method cannot properly support per-subscriber filtering, and may \
+            return the `Id` of a disabled span if per-subscriber filtering is in \
+            use. use `.parent().map(SpanRef::id)` instead.",
+        since = "0.2.21"
+    )]
     pub fn parent_id(&self) -> Option<&Id> {
+        // XXX(eliza): this doesn't work with PLF because the ID is potentially
+        // borrowed from a parent we got from the registry, rather than from
+        // `self`, so we can't return a borrowed parent. so, right now, we just
+        // return the actual parent ID, and ignore PLF. which is not great.
+        //
+        // i think if we want this to play nice with PLF, we should just change
+        // it to return the `Id` by value instead of `&Id` (which we ought to do
+        // anyway since an `Id` is just a word) but that's a breaking change.
+        // alternatively, we could deprecate this method since it can't support
+        // PLF in its current form (which is what we would want to do if we want
+        // to release PLF in a minor version)...
```

Comment:
```suggestion
        // XXX(eliza): this doesn't work with PSF because the ID is potentially
        // borrowed from a parent we got from the registry, rather than from
        // `self`, so we can't return a borrowed parent. so, right now, we just
        // return the actual parent ID, and ignore PSF. which is not great.
        //
        // i think if we want this to play nice with PSF, we should just change
        // it to return the `Id` by value instead of `&Id` (which we ought to do
        // anyway since an `Id` is just a word) but that's a breaking change.
        // alternatively, we could deprecate this method since it can't support
        // PSF in its current form (which is what we would want to do if we want
        // to release PSF in a minor version)...
```